csharp-linq/LINQInnerJoin/LinqInnerJoin/EnrolmentHelper.cs

```diff
@@ -0,0 +1,209 @@
+﻿namespace LinqInnerJoin
+{
+    public static class EnrolmentHelper
+    {
+        public static IEnumerable<dynamic> GetEnrolments(
+            IEnumerable<Enrolment> enrolments)
+        {
+            var students = Student.GetDummyStudents();
+
+            var courses = Course.GetDummyCourses();
+
+            var enrolmentsRelation = GetEnrolmentsRelation(
+                enrolments,
+                students,
+                courses);
+
+            var result = enrolmentsRelation.Select(
+                e =>
+                    new
+                    {
+                        e.enrolmentStudentRelation.enrolment.Id,
+                        StudentName = e.enrolmentStudentRelation.student.Name,
+                        CourseName = e.course.Name,
+                    }
+            );
+
+            return result;
+        }
+
+        public static IEnumerable<dynamic> FilterEnrolments(IEnumerable<Enrolment> enrolments)
+        {
+            var students = Student.GetDummyStudents();
+
+            var courses = Course.GetDummyCourses();
+
+            var enrolmentsRelation = GetEnrolmentsRelation(enrolments, students, courses);
+
+            var result = enrolmentsRelation
+                .Select(
+                    e =>
+                        new
+                        {
+                            e.enrolmentStudentRelation.enrolment.Id,
+                            StudentName = e.enrolmentStudentRelation.student.Name,
+                            CourseId = e.course.Id,
+                            CourseName = e.course.Name,
+                        }
+                )
+                .Where(x => x.CourseId == 1);
+
+            return result;
+        }
+
+        public static IEnumerable<dynamic> FilterAndGroupEnrolments(
+            IEnumerable<Enrolment> enrolments)
+        {
+            var students = Student.GetDummyStudents();
+
+            var courses = Course.GetDummyCourses();
+
+            var enrolmentsRelation = GetEnrolmentsRelation(
+                enrolments,
+                students,
+                courses);
+
+            var result = enrolmentsRelation
+                .Select(
+                    e =>
+                        new
+                        {
+                            e.enrolmentStudentRelation.enrolment.Id,
+                            StudentName = e.enrolmentStudentRelation.student.Name,
+                            CourseId = e.course.Id,
+                            CourseName = e.course.Name,
+                        }
+                )
+                .Where(x => x.CourseId < 5)
+                .GroupBy(x => x.CourseId);
+
+            return result;
+        }
+
+        public static IEnumerable<dynamic> GetCoursesWithCategory(
+            IEnumerable<Course> courses)
+        {
+            var catrgories = Category.GetDummyCourseCategories();
```

Comment:
type in "catrgories" instead of "categories"

Reply:
done